turbinia/lib/docker_manager.py

```diff
@@ -0,0 +1,222 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Library to handle docker related queries."""
+
+from __future__ import unicode_literals
+
+import logging
+import codecs
+import os
+import stat
+import docker
+
+from turbinia import TurbiniaException
+
+log = logging.getLogger('turbinia')
+
+
+def IsBlockDevice(path):
+  """Checks path to determine whether it is a block device.
+
+  Args:
+      path: String of path to check.
+
+  Returns:
+      Bool indicating success.
+  """
+  if not os.path.exists(path):
+    return False
+  mode = os.stat(path).st_mode
+  return stat.S_ISBLK(mode)
+
+
+class DockerManager(object):
+  """Class handling Docker management."""
+
+  def __init__(self):
+    """Initialize the Docker manager client."""
+    self.client = self._create_client()
+
+  def _create_client(self):
+    """Creates a Docker client.
+
+    Initializes a connection to the Docker daemon using
+    preconfigured Docker environment variables.
+
+    Returns:
+      docker_client(DockerClient): The Docker daemon client.
+
+    Raises:
+      TurbiniaException: If the Docker daemon could not be connected to.
+    """
+    try:
+      docker_client = docker.from_env()
+    except docker.errors.APIError as exception:
+      raise TurbiniaException(
+          'An issue has occured connecting to the Docker daemon: {0!s}'.format(
+              exception))
+    return docker_client
+
+  def verify_image(self, image_id):
```

Comment:
Should this maybe be `get_image` instead of `verify_image` since you're really just getting an image object from the id?

Reply:
Looking at the other code, I'm not sure if this actually returns an image object or an id, so this may or may not be a relevant comment.

Reply:
ok, sounds good, thanks.